packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("Jul 17, 2020 @ Jul 16, 2020")
+			.realPress("Tab");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value", "Jul 16, 2020 @ Jul 17, 2020");
+	});
+
+	it("Delimiter is part of the format pattern", () => {
+		cy.mount(<DateRangePicker formatPattern="yyyy-MM-dd"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("2020-09-09 - 2020-10-10")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value-state", "None");
+	});
+
+	it("Month name changes on next button press", () => {
+		cy.mount(<DateRangePicker value="09/09/2020 - 10/10/2020" formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.as("calendar")
+			.shadow()
+			.find(".ui5-calheader")
+			.as("calendarHeader");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@calendarHeader")
+			.find("[data-ui5-cal-header-btn-next]")
+			.realClick();
+
+		cy.get("@calendarHeader")
+			.find("[data-ui5-cal-header-btn-month]")
+			.as("monthButton")
+			.should("have.text", "October");
+	});
+
+	it("startDateValue and endDateValue getters when only start date is present", () => {
+		cy.mount(<DateRangePicker value="27/09/2019" formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.then($datePicker => {
+				cy.wrap({
+					startDateValue: () => $datePicker[0].startDateValue,
+					endDateValue: () => $datePicker[0].endDateValue
+				})
+					.as("dates")
+					.invoke("startDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp)).to.deep.equal(new Date(2019, 8, 27));
+					});
+
+				cy.get("@dates")
+					.invoke("endDateValue")
+					.then(date => {
+						expect(date.toString()).to.be.equal("Invalid Date");
+					});
+			});
+	});
+
+	it("Picker popover should have accessible name", () => {
+		cy.mount(<DateRangePicker></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-responsive-popover")
+			.should("have.attr", "accessible-name", "Choose Date Range");
+	});
+
+	it("Selected days: accessibility semantics", () => {
+		cy.wrap({ setLanguage })
+			.invoke("setLanguage", "en");
+
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/06/2024 - 15/06/2024")
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type